EasyPost/Client.cs

```diff
@@ -12,18 +12,27 @@
 
 namespace EasyPost {
     public class Client {
+
+        internal const string ApiBaseUrl = "https://api.easypost.com/v2";
+
         public static string apiKey { get; set; }
         public static string apiBase { get; set; }
 
         public string version;
 
         internal RestClient client;
+        internal ClientConfiguration configuration;
 
-        public Client(string apiBaseUrl = "https://api.easypost.com/v2") {
+        public Client(string apiBaseUrl = ApiBaseUrl)
+            : this(new ClientConfiguration(apiKey, apiBase ?? apiBaseUrl)) { }
+
+        public Client(ClientConfiguration clientConfiguration) {
             System.Net.ServicePointManager.SecurityProtocol = Security.GetProtocol();
 
-            apiBaseUrl = apiBase ?? apiBaseUrl;
-            client = new RestClient(apiBaseUrl);
+            if (clientConfiguration == null) throw new ArgumentNullException(nameof(clientConfiguration));
```

Comment:
This is introduced in .NET 4.6, this library was written with 4.5. Is there an equivalent function in 4.5?